src/message_stream.rs

```diff
@@ -153,7 +153,13 @@ where
         &mut self,
     ) -> Result<Option<MessageEvent<<A::Handler as Decode>::Item>>> {
         loop {
-            track!(self.rbuf.fill(&mut self.transport_stream))?;
+            match track!(self.rbuf.fill(&mut self.transport_stream)) {
+                Err(_e) => {
```

Comment:
I think that ignoring all errors may be dangerous (they may include critical errors, but users cannot notice them).
Besides, in your implementation, the stream encountered a error may become a zombie (it is never polled but alive because it had not returned `Ok(Async::Ready(None))` to the caller).

I tried to fix this problem as the following patch. Could you confirm this patch works for you?
```diff
diff --git a/src/message_stream.rs b/src/message_stream.rs
index ba27872..edb116c 100644
--- a/src/message_stream.rs
+++ b/src/message_stream.rs
@@ -7,6 +7,7 @@ use fibers::sync::mpsc;
 use fibers::time::timer::{self, Timeout};
 use fibers_tasque::DefaultCpuTaskQueue;
 use futures::{Async, Future, Poll, Stream};
+use std;
 use std::cmp;
 use std::collections::{BinaryHeap, HashMap};
 use std::fmt;
@@ -153,7 +154,17 @@ where
         &mut self,
     ) -> Result<Option<MessageEvent<<A::Handler as Decode>::Item>>> {
         loop {
-            track!(self.rbuf.fill(&mut self.transport_stream))?;
+            if let Err(e) = track!(self.rbuf.fill(&mut self.transport_stream)) {
+                // In Windows environment, a `ConnectionReset` error may be issued even
+                // if the client closed the connection normally.
+                // Since the error is noisy and usually not critical one, we simply ignore it.
+                let is_ignorable_error = e
+                    .concrete_cause::<std::io::Error>()
+                    .map_or(false, |e| e.kind() == std::io::ErrorKind::ConnectionReset);
+                if !is_ignorable_error {
+                    return Err(Error::from(e));
+                }
+            }
 
             if !self.packet_header_decoder.is_idle() {
                 track!(self
@@ -306,7 +317,7 @@ where
             "This stream may be overloaded"
         );
 
-        if self.rbuf.stream_state().is_eos() {
+        if self.rbuf.stream_state().is_eos() || self.rbuf.stream_state().is_error() {
             Ok(Async::Ready(None))
         } else {
             Ok(Async::NotReady)
```